.github/workflows/test_and_build_image.yaml

```diff
@@ -25,13 +25,13 @@ jobs:
           go-version: 1.19
         id: go
       - name: Check out code into the Go module directory
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
         if: github.event_name == 'pull_request'
         with:
           ref: ${{ github.event.pull_request.head.sha }}
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
         if: github.event_name != 'pull_request'
-      - uses: actions/cache@v2
+      - uses: actions/cache@v3
```

Comment:
uses: actions/setup-go@v3 can also do caching:

```yaml
    - name: Set up Go
      uses: actions/setup-go@v3
      with:
        go-version-file: 'go.mod'
        cache: true
```

Thus, the action/cache can be removed